tests/orchestrations_test.go

```diff
@@ -715,6 +715,142 @@ func Test_TerminateOrchestration(t *testing.T) {
 	)
 }
 
+func Test_TerminateOrchestration_Recursive(t *testing.T) {
+	delayTime := 4 * time.Second
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("Root", func(ctx *task.OrchestrationContext) (any, error) {
+		ctx.CallSubOrchestrator("L1", task.WithSubOrchestrationInstanceID(string(ctx.ID)+"_L1")).Await(nil)
+		return nil, nil
+	})
+	r.AddOrchestratorN("L1", func(ctx *task.OrchestrationContext) (any, error) {
+		ctx.CallSubOrchestrator("L2", task.WithSubOrchestrationInstanceID(string(ctx.ID)+"_L2")).Await(nil)
+		return nil, nil
+	})
+	r.AddOrchestratorN("L2", func(ctx *task.OrchestrationContext) (any, error) {
+		ctx.CreateTimer(delayTime).Await(nil)
```

Comment:
I noticed that you left out some important validations that were in the original recursive terminate test in grpc_test.go - for example, validating that the orchestrations actually stop and don't execute the activity function at the leaf. Can you please use that as the basis for this test, and make just additive changes rather than removing validations?

Reply:
Added.